community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexReader.java

```diff
@@ -30,86 +30,79 @@
     {
-        this.numberReader = numberReader;
-        this.spatialReader = spatialReader;
-        this.temporalReader = temporalReader;
-        this.luceneReader = luceneReader;
-        this.selector = selector;
+        super( readers, selector );
         this.descriptor = descriptor;
     }
 
     @Override
     public void close()
     {
-        forAll( Resource::close, numberReader, spatialReader, temporalReader, luceneReader );
+        forAll( Resource::close, instances );
     }
 
     @Override
     public long countIndexedNodes( long nodeId, Value... propertyValues )
     {
-        return selector.select( numberReader, spatialReader, temporalReader, luceneReader, propertyValues ).countIndexedNodes( nodeId, propertyValues );
+        return selector.select( instances, propertyValues ).countIndexedNodes( nodeId, propertyValues );
     }
 
     @Override
     public IndexSampler createSampler()
     {
-        return new FusionIndexSampler(
-                numberReader.createSampler(),
-                spatialReader.createSampler(),
-                temporalReader.createSampler(),
-                luceneReader.createSampler() );
+        return new FusionIndexSampler( instancesAs( IndexSampler.class, IndexReader::createSampler ) );
     }
 
     @Override
     public PrimitiveLongResourceIterator query( IndexQuery... predicates ) throws IndexNotApplicableKernelException
     {
         if ( predicates.length > 1 )
         {
-            return luceneReader.query( predicates );
+            return instances[LUCENE].query( predicates );
         }
+        IndexQuery predicate = predicates[0];
 
-        if ( predicates[0] instanceof ExactPredicate )
+        if ( predicate instanceof ExactPredicate )
         {
             ExactPredicate exactPredicate = (ExactPredicate) predicates[0];
-            return selector.select( numberReader, spatialReader, temporalReader, luceneReader, exactPredicate.value() ).query( predicates );
+            return selector.select( instances, exactPredicate.value() ).query( predicates );
+        }
+
+        if ( predicate instanceof NumberRangePredicate )
+        {
+            return instances[NUMBER].query( predicates );
         }
 
-        if ( predicates[0] instanceof NumberRangePredicate )
+        if ( predicate instanceof StringRangePredicate ||
+             predicate instanceof StringPrefixPredicate ||
+             predicate instanceof StringSuffixPredicate ||
+             predicate instanceof StringContainsPredicate )
         {
-            return numberReader.query( predicates );
+            return instances[STRING].query( predicate );
         }
 
-        if ( predicates[0] instanceof GeometryRangePredicate )
+        if ( predicate instanceof GeometryRangePredicate )
         {
-            return spatialReader.query( predicates );
+            return instances[SPATIAL].query( predicates );
         }
 
 // TODO: support temporal range queries
```

Comment:
do you guys know when/who will gonna fix those todos or we can remove them?

Reply:
This TODO is from code that was just moved around in this PR. Pretty sure there's a PR up now (Temporal IV) that will implement those

Reply:
ok